spec/controllers/courses_controller_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'rails_helper'
+
+RSpec.describe CoursesController, :type => :controller do
+  let!(:course) { create(:course, name: "Criminology", allocation: 20) }
+
+  describe "GET #index" do
+    subject { get :index }
+
+    it "populates an array of students" do
+      subject
+      expect(assigns(:courses)).to eq([course])
+    end
+
+    it { is_expected.to have_http_status(:success) }
+    it { is_expected.to render_template("courses/index") }
+  end
+
+  describe "GET #show" do
+    subject { get :show, id: course }
+
+    it "assigns the requests course to @course" do
+      subject
+      expect(assigns(:course)).to eq(course)
+    end
+
+    it { is_expected.to have_http_status(:success) }
+    it { is_expected.to render_template("courses/show") }
+  end
+
+  describe "GET #edit" do
+    subject { get :edit, id: course }
+
+    it "assigns the requests course to @course" do
+      subject
+      expect(assigns(:course)).to eq(course)
+    end
+
+    it { is_expected.to have_http_status(:success) }
+    it { is_expected.to render_template("courses/edit") }
+  end
+
+  describe "PUT #update" do
+    context "valid attributes" do
+      subject { put :update, id: course, course: attributes_for(:course, name: "Sociology", allocation: 30) }
+
+      it "located the requested @course" do
+        subject
+        expect(assigns(:course)).to eq(course)
+      end
+
+      it "changes @course's attributes" do
+        subject
+        expect { course.reload }.to change { course.name }.to("Sociology")
```

Comment:
this should work with `expect { subject }`

```suggestion
        expect { subject }.to change { course.name }.to("Sociology")
```

Reply:
Need to call `reload` on the course object to trigger the change.

Reply:
Ah yea.. 2 ways to do this

(preferred):

```ruby
.to change { course.reload.name}
```

or

```ruby
.to change { Course.last.name }
```